routes/index.js

```diff
@@ -2,20 +2,28 @@ var express = require('express');
 var router = express.Router();
 
 let index = require('../controllers/index');
-let logIn = require('../controllers/login');
-let signUp = require('../controllers/signup')
+let loginOut = require('../controllers/login-out');
+let signUp = require('../controllers/signup');
+let handleAuth = require('../public/javascripts/handleAuth')
+
 
 // home page
-router.get('/', index.index);
-router.post('/', index.add_paragraph);
+// restrict access to users not logged in
+router.get('/', handleAuth.notAuth, handleAuth.currentUser, index.index);
+router.post('/', handleAuth.currentUser, index.add_paragraph);
```

Comment:
Really nice job using the middleware functions here to scope access to your routes